### PR TITLE
fix(core): make health check http endpoint responsive when memory limit is reached

### PR DESCRIPTION
**Pull request by ideoma**
Fixes https://github.com/questdb/questdb/issues/4999

Changes:

- health check HTTP server now has separate config values to keep buffer sizes independent from the main HTTP server and low by default
- health check HTTP server processing context does not release memory when pooled so that processing the incoming requests does not need to allocate the memory

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 94 / 99 (94.95%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/WeakMutableObjectPool.java|2|5|40.00%|
|:large_blue_circle:|io/questdb/std/Unsafe.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|33|34|97.06%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpResponseSink.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropHttpServerConfiguration.java|38|38|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 94 / 99 (94.95%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/WeakMutableObjectPool.java|2|5|40.00%|
|:large_blue_circle:|io/questdb/std/Unsafe.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|33|34|97.06%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpResponseSink.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropHttpServerConfiguration.java|38|38|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 94 / 99 (94.95%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/WeakMutableObjectPool.java|2|5|40.00%|
|:large_blue_circle:|io/questdb/std/Unsafe.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|33|34|97.06%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpResponseSink.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropHttpServerConfiguration.java|38|38|100.00%|

**Review by bluestreak01 — APPROVED** [review]

File: core/src/main/java/io/questdb/cutlass/http/HttpConnectionContext.java [hunk 1]
```diff
@@ -124,6 +125,11 @@ public HttpConnectionContext(
         this.multipartContentParser = new HttpMultipartContentParser(multipartContentHeaderParser);
         this.responseSink = new HttpResponseSink(contextConfiguration);
         this.recvBufferSize = contextConfiguration.getRecvBufferSize();
+        this.preAllocateBuffers = configuration.isHealthCheck();
```
[the remark this answers is not in the document]
Comment (ideoma) [reply core/src/main/java/io/questdb/cutlass/http/HttpConnectionContext.java]: done

**Review by bluestreak01 — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 94 / 99 (94.95%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/WeakMutableObjectPool.java|2|5|40.00%|
|:large_blue_circle:|io/questdb/std/Unsafe.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|33|34|97.06%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpResponseSink.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropHttpServerConfiguration.java|38|38|100.00%|

**Pull request merged**